src/core/Akka.Remote/RemoteActorRefProvider.cs

```diff
@@ -435,7 +435,7 @@ public Deploy LookUpRemotes(IEnumerable<string> p)
 
         public bool HasAddress(Address address)
         {
-            return address == _local.RootPath.Address || address == RootPath.Address || Transport.Addresses.Any(a => a == address);
+            return address == _local.RootPath.Address || Transport.Addresses.Contains(address);
```

Comment:
Ah I see what you meant in your other comment

Reply:
would interest me to make a banchmark on it.

if the volatile fields and the removal of the AddressCache had an impact